tensorflow_model_optimization/python/core/clustering/keras/cluster_integration_test.py

```diff
@@ -160,6 +184,105 @@ def clusters_check(stripped_model):
 
     self.end_to_end_testing(original_model, clusters_check)
 
+  @keras_parameterized.run_all_keras_modes(always_skip_v1=True)
+  def testEndToEndSubclassedModel(self):
+    """Test End to End clustering for the subclassed model.
+    In this test we pass the whole subclassed model for clustering.
+    We check that the number of weights is less the requested
+    number of clusters after stripping clustering wrapper.
+
+    """
+    subclassed_model = SubclassedModel()
+
+    clustered_model = cluster.cluster_weights(subclassed_model, **self.params)
+
+    clustered_model.compile(
+        loss=keras.losses.categorical_crossentropy,
+        optimizer="adam",
+        metrics=["accuracy"]
+    )
+
+    # The model should be trained a little bit.
+    clustered_model.fit(x=self.dataset_generator(), steps_per_epoch=1)
+    stripped_model = cluster.strip_clustering(clustered_model)
+
+    nr_unique_weights = len(np.unique(stripped_model.layers[0].\
+        trainable_weights[0].numpy().flatten()))
+    self.assertLessEqual(nr_unique_weights, self.params["number_of_clusters"])
+
+  @keras_parameterized.run_all_keras_modes(always_skip_v1=True)
+  def testEndToEndSubclassedModelTwoLayers(self):
```

Comment:
This test re-produces the approach tested here: https://github.com/tensorflow/model-optimization/pull/554